tests/test_import.py

```diff
@@ -83,3 +95,27 @@ def test_import_histosys():
     }
 
     assert channels['channel2']['samples'][0]['modifiers'][0]['type'] == 'histosys'
+
+
+def test_import_filecache(mocker):
+
+    # uproot is dynamically imported in readxml, so to wrap it we need to set it now
+    pyhf.readxml.uproot = uproot
+
```

Comment:
would it help if we moved the uproot import to the module-level? it's imported dynamically in order to only do the import incase someone explicitly wants xml2json (if they already have json there's no need to install uproot) but we could easily just do 

```
try:
    import readxml
except ImportError
    pass # I guess we're not importing
```

in commandline.py (or factor xml2json into a separate module)